algo/src/main/java/org/neo4j/graphalgo/impl/MSTPrim.java

```diff
@@ -0,0 +1,90 @@
+package org.neo4j.graphalgo.impl;
+
+import com.carrotsearch.hppc.*;
+import org.neo4j.graphalgo.api.Graph;
+import org.neo4j.graphalgo.api.IntBinaryConsumer;
+import org.neo4j.graphalgo.core.utils.LongMinPriorityQueue;
+import org.neo4j.graphalgo.core.utils.RawValues;
+import org.neo4j.graphdb.Direction;
+
+import static org.neo4j.graphalgo.core.utils.RawValues.*;
+
+/**
+ * Impl. Prim's Minimum Weight Spanning Tree
+ *
+ * @author mknobloch
+ */
+public class MSTPrim {
+
+    private final Graph graph;
+
+    public MSTPrim(Graph graph) {
+        this.graph = graph;
+    }
+
+    /**
+     * compute the minimum weight spanning tree starting at node startNode
+     *
+     * @param startNode the node to start the evaluation from
+     * @return a container of the transitions in the minimum spanning tree
+     */
+    public MST compute(int startNode) {
+
+        final LongMinPriorityQueue queue = new LongMinPriorityQueue();
+        final MST mst = new MST(graph.nodeCount());
+        final BitSet visited = new BitSet(graph.nodeCount());
+
+        // initially add all relations from startNode to the priority queue
+        visited.set(startNode);
+        graph.forEachRelation(startNode, Direction.BOTH, (sourceNodeId, targetNodeId, relationId, weight) -> {
```

Comment:
I'm just concerned that we're doing double the work, b/c one nodes outgoing relationship is the other nodes incoming one. But I trust you on this.